pkg/actuators/machine/instances.go

```diff
@@ -801,3 +819,21 @@ func correctExistingTags(machine *machinev1beta1.Machine, regionID string, insta
 
 	return nil
 }
+
+// getResourceGroupId takes an AlibabaCloudMachineProviderConfig and will return the
+// resource group id if available, or determine the group id by using the search tags.
+// An error will be returned if no group id can be found, or if multiple groups are
+// found from the search tags.
+func getResourceGroupId(machineProviderConfig *machinev1.AlibabaCloudMachineProviderConfig) (string, error) {
+	switch machineProviderConfig.ResourceGroup.Type {
```

Comment:
it seems like we fail if the resource group is nil currently, the way the code is structured i'm not sure how it's supposed to find the security groups without the group id. but maybe that's ok?

Reply:
i asked around a little bit, and we should not allow creation of resources in the default group. the user should always specify the resource group.

Reply:
I saw a comment about most customers using the default group, are we sure this is correct?

Reply:
+1, we agreed that the user should always provide this information, even in the case where they want to use the default resource group. this will help ensure that the user isn't creating resources in the wrong group by accident.